### ISSUE TITLE
Some questions for the KL_loss and KL_loss_r and model behaviors

### ISSUE THREAD

**Issue by feng-yufei** [PlayVoice/vits_chinese#34]
Hello MaxMax2016, 

Thank you for sharing your code on the improved VITS, I hope to check with you about the model behaviors when adding the bi-directional KL divergence. In this repo I fubd you use 1.0 * kl_loss + 1.0 * kl_loss_r during training, and when I do training on my own dataset I saw this greatly increased the mel-spectrogram loss, comparing to the original VITS, and when I add kl_loss_r to finetune a well trained VITS model, it gives bad voice quality.
Can you share your experiences or findings when trying to add this specific term?

Thanks

**Comment by MaxMax2016** [comment]
you can freeze post encoder and hifigan decoder，maybe also text encoder, then use add kl_loss_r finetune a well trained VITS model.

**Comment by MaxMax2016** [comment]
> 关注下，这个loss 加上有效果不，有过对比吗

with this loss, less sound error than raw model.

**Comment by GuangChen2016** [comment]
@MaxMax2016 加上这个loss后，有比较高add_blank和不加add_blank以及是不是用sdp duration的差异么

**Comment by feng-yufei** [comment]
> you can freeze post encoder and hifigan decoder，maybe also text encoder, then use add kl_loss_r finetune a well trained VITS model (train flow).

Does your kl_loss_r term increase the mel_spectrogram loss during training (training from scratch)? If you also observed this (as I did), does it affect the voice quality after convergence?

**Comment by MaxMax2016** [comment]
i have not meet this problem, because baker is high precision labeled. I wonder, if the data is labeled incorrectly, kl_loss_r will propagate error to subsequent modules(post encoder and hifigan decoder), so the best way is :you can freeze post encoder and hifigan decoder，maybe also text encoder, then use add kl_loss_r finetune a well trained VITS model (train flow). @feng-yufei

**Comment by MaxMax2016** [comment]
@feng-yufei https://github.com/PlayVoice/vits_chinese/issues/36#issuecomment-1463298450  YES it increases mel loss.

**Comment by GuangChen2016** [comment]
@feng-yufei Hi, I met the same issues using my datasets, what's your conclusions now?

**Comment by GuangChen2016** [comment]
@feng-yufei Yeah, and did you use loss_kl_r now and set the coefficient to 0.01 for your training or just remove  loss_kl_r?

**Comment by iooops** [comment]
> @MaxMax2016 加上这个loss后，有比较高add_blank和不加add_blank以及是不是用sdp duration的差异么

请问加add_blank和不加add_blank有啥差异么？
